### PR TITLE
[ci] Define machines to run the CI with Jenkins params

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry all

File: Jenkinsfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 def dirPrefix = 'reframe-ci'
 def loginBash = '#!/bin/bash -l'
 def bashScript = 'ci-scripts/ci-runner.bash'
-def machinesList = ['daint', 'dom', 'kesch']
+def machinesList = params.machines.split()
```
Comment (vkarak) [inline Jenkinsfile]: Where these `params` are gotten from? From the Jenkins project? Could you then remove `Kesch` from there?

Comment (teojgo) [reply Jenkinsfile]: The `params` is injected by Jenkins and if the build is not triggered manually the parameters have the default values.

**Comment by teojgo** [comment]
@jenkins-cscs retry all

File: Jenkinsfile (hunk 1 above)
Comment (vkarak) [reply Jenkinsfile]: So the defaults are set by the Jenkins project?

Comment (teojgo) [reply Jenkinsfile]: You add a parameter and give it default values yes.

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**